src/library.h

```diff
@@ -51,6 +52,7 @@ public slots:
     void bookmarksChanged();
 
 private:
+    QMutex mutex;
```

Comment:
Would be better to give a more useful name. Something like `m_updateFromDirMutex`

Reply:
Done